### PR TITLE
Replace deprecated API/Class with new API/Class in engine code and test cases.

### PR DESCRIPTION
**Pull request by WenhaiLin**

### CODE REVIEW

File: cocos/base/ccUTF8.cpp [hunk 1]
```diff
@@ -31,6 +31,27 @@
...
+        
+    char* buf = (char*)malloc(CC_MAX_STRING_LENGTH);
+    if (buf != nullptr)
+    {
+        vsnprintf(buf, CC_MAX_STRING_LENGTH, format, ap);
+        ret = buf;
+        free(buf);
+    }
+    va_end(ap);
+
```
Comment (minggo) [inline cocos/base/ccUTF8.cpp]: Please undefine here.

File: cocos/editor-support/cocosbuilder/CCNodeLoader.cpp [hunk 2]
```diff
@@ -96,14 +100,13 @@ void NodeLoader::parseProperties(Node * pNode, Node * pParent, CCBReader * ccbRe
         }
         else if (isExtraProp && pNode == ccbReader->getAnimationManager()->getRootNode())
         {
-            __Array *extraPropsNames = static_cast<__Array*>(pNode->getUserObject());
+            auto extraPropsNames = (__StringArray*)(pNode->getUserObject());
             if (! extraPropsNames)
             {
-                extraPropsNames = Array::create();
```
Comment (zilongshanren) [inline cocos/editor-support/cocosbuilder/CCNodeLoader.cpp:102]: The new statement will cause memory leaks.

File: cocos/scripting/lua-bindings/manual/cocos2d/lua_cocos2dx_manual.cpp [hunk 3]
```diff
@@ -32,6 +32,36 @@
 #include <sys/socket.h>
 #endif
 
+class __TouchListenerHandler : public Map<const char*, Ref*>, public Ref
```
Comment (zilongshanren) [inline cocos/scripting/lua-bindings/manual/cocos2d/lua_cocos2dx_manual.cpp]: We should avoid multiple inheritance. It is very weird.

File: cocos/editor-support/cocosbuilder/CCNodeLoader.cpp [hunk 4]
```diff
@@ -11,6 +11,12 @@ using namespace std;
 using namespace cocos2d;
 using namespace cocos2d::extension;
 
+class __StringArray : public Ref
```
Comment (minggo) [inline cocos/editor-support/cocosbuilder/CCNodeLoader.cpp]: __StringArray -> StringArray

Comment (minggo) [inline cocos/editor-support/cocosbuilder/CCNodeLoader.cpp]: And it is better to put it into a anonymous namespace.

**Pull request merged**